includes/Modules/Analytics.php

```diff
@@ -933,7 +933,15 @@ function ( $dimension_def ) {
 				$start_date  = $data['startDate'];
 				$end_date    = $data['endDate'];
 				if ( strtotime( $start_date ) && strtotime( $end_date ) ) {
-					$date_ranges[] = array( $start_date, $end_date );
+					$compare_start_date = $data['compareStartDate'];
+					$compare_end_date   = $data['compareEndDate'];
+					$date_ranges[]      = array( $start_date, $end_date );
+
+					// When using multiple date ranges, it changes the structure of the response,
+					// where each date range becomes an item in a list.
+					if ( strtotime( $compare_start_date ) && strtotime( $compare_end_date ) ) {
+						array_push( $date_ranges, $compare_start_date, $compare_end_date );
```

Comment:
This is not correct, the compare start and end dates should be the first and second element of its own array, and that array should then be the second item in `$date_ranges` (see code in the else clause below).

Also nit-picking, but let's not use `array_push( $date_ranges, ... )`, but rather `$date_ranges[] = ...`.
```suggestion
						$date_ranges[] = array( $compare_start_date, $compare_end_date );
```